File: testdata/fixtures/js_clones/userService.ts
import { db } from './db';
import { NotFoundError } from './errors';

interface User {
  id: string;
  name: string;
  email: string;
}

export async function getUser(id: string): Promise<User> {
  const user = await db.query('SELECT * FROM users WHERE id = $1', [id]);
  if (!user) {
    throw new NotFoundError(`User not found: ${id}`);
  }
  if (!user.active) {
    throw new Error('User is inactive');
  }
  return {
    id: user.id,
    name: user.name,
    email: user.email,
  };
}

export async function updateUser(id: string, data: Partial<User>): Promise<User> {
  const user = await db.query('SELECT * FROM users WHERE id = $1', [id]);
  if (!user) {
    throw new NotFoundError(`User not found: ${id}`);
  }
  if (!user.active) {
    throw new Error('User is inactive');
  }
  const updated = await db.query('UPDATE users SET name = $1, email = $2 WHERE id = $3 RETURNING *', [
    data.name || user.name,
    data.email || user.email,
    id,
  ]);
  return {
    id: updated.id,
    name: updated.name,
    email: updated.email,
  };
}

export async function deleteUser(id: string): Promise<void> {
  const user = await db.query('SELECT * FROM users WHERE id = $1', [id]);
  if (!user) {
    throw new NotFoundError(`User not found: ${id}`);
  }
  if (!user.active) {
    throw new Error('User is inactive');
  }
  await db.query('DELETE FROM users WHERE id = $1', [id]);
}

File: testdata/fixtures/rb_clones/user_service.rb
class UserService
  def initialize(repo)
    @repo = repo
  end

  def find_user(id)
    user = @repo.find(id)
    raise NotFoundError, "User not found: #{id}" unless user
    raise InactiveError, "User is inactive" unless user.active?
    user
  end

  def update_user(id, attrs)
    user = @repo.find(id)
    raise NotFoundError, "User not found: #{id}" unless user
    raise InactiveError, "User is inactive" unless user.active?
    user.update(attrs)
    @repo.save(user)
    user
  end

  def delete_user(id)
    user = @repo.find(id)
    raise NotFoundError, "User not found: #{id}" unless user
    raise InactiveError, "User is inactive" unless user.active?
    @repo.delete(user)
  end
end
